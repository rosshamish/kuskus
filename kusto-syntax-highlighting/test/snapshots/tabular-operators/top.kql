// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/topoperator

T | top 5 by Name desc nulls last

// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/topnestedoperator

StormEvents
| top-nested 2 of State with others = "All Other States" by sum(BeginLat),
  top-nested 3 of Source by sum(BeginLat),
  top-nested 1 of EndLocation with others = "All Other End Locations" by sum(BeginLat)

// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/tophittersoperator

Events | top-hitters 5 of EventId

PageViews | top-hitters 25 of Page by NumViews
