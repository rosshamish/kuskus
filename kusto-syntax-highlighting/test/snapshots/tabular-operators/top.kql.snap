>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/topoperator
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>T | top 5 by Name desc nulls last
#^^^^ source.kusto
#    ^^^ source.kusto meta.query.top.kusto keyword.other.query.kusto
#       ^ source.kusto meta.query.top.kusto
#        ^ source.kusto meta.query.top.kusto constant.numeric.kusto
#         ^ source.kusto meta.query.top.kusto
#          ^^ source.kusto meta.query.top.kusto keyword.other.operator.kusto
#            ^ source.kusto meta.query.top.kusto
#             ^^^^ source.kusto meta.query.top.kusto variable.other.column.kusto
#                 ^ source.kusto
#                  ^^^^ source.kusto keyword.other.kusto
#                      ^ source.kusto
#                       ^^^^^^^^^^ source.kusto keyword.other.kusto
>
>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/topnestedoperator
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>StormEvents
#^^^^^^^^^^^^ source.kusto
>| top-nested 2 of State with others = "All Other States" by sum(BeginLat),
#^^ source.kusto
#  ^^^^^^^^^^ source.kusto keyword.other.query.kusto
#            ^ source.kusto
#             ^ source.kusto constant.numeric.kusto
#              ^ source.kusto
#               ^^ source.kusto keyword.other.operator.kusto
#                 ^^^^^^^ source.kusto
#                        ^^^^ source.kusto keyword.other.operator.kusto
#                            ^^^^^^^^^^ source.kusto
#                                      ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                       ^^^^^^^^^^^^^^^^ source.kusto string.quoted.double.kusto
#                                                       ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                                        ^ source.kusto
#                                                         ^^ source.kusto keyword.other.operator.kusto
#                                                           ^ source.kusto
#                                                            ^^^ source.kusto support.function.kusto
#                                                               ^^^^^^^^^^^^ source.kusto
>  top-nested 3 of Source by sum(BeginLat),
#^^ source.kusto
#  ^^^^^^^^^^ source.kusto keyword.other.query.kusto
#            ^ source.kusto
#             ^ source.kusto constant.numeric.kusto
#              ^ source.kusto
#               ^^ source.kusto keyword.other.operator.kusto
#                 ^^^^^^^^ source.kusto
#                         ^^ source.kusto keyword.other.operator.kusto
#                           ^ source.kusto
#                            ^^^ source.kusto support.function.kusto
#                               ^^^^^^^^^^^^ source.kusto
>  top-nested 1 of EndLocation with others = "All Other End Locations" by sum(BeginLat)
#^^ source.kusto
#  ^^^^^^^^^^ source.kusto keyword.other.query.kusto
#            ^ source.kusto
#             ^ source.kusto constant.numeric.kusto
#              ^ source.kusto
#               ^^ source.kusto keyword.other.operator.kusto
#                 ^^^^^^^^^^^^^ source.kusto
#                              ^^^^ source.kusto keyword.other.operator.kusto
#                                  ^^^^^^^^^^ source.kusto
#                                            ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                             ^^^^^^^^^^^^^^^^^^^^^^^ source.kusto string.quoted.double.kusto
#                                                                    ^ source.kusto string.quoted.double.kusto punctuation.definition.string.kusto
#                                                                     ^ source.kusto
#                                                                      ^^ source.kusto keyword.other.operator.kusto
#                                                                        ^ source.kusto
#                                                                         ^^^ source.kusto support.function.kusto
#                                                                            ^^^^^^^^^^^ source.kusto
>
>// https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/tophittersoperator
#^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.kusto comment.line.kusto
>
>Events | top-hitters 5 of EventId
#^^^^^^^^^ source.kusto
#         ^^^^^^^^^^^ source.kusto meta.query.top-hitters.kusto keyword.other.query.kusto
#                    ^ source.kusto meta.query.top-hitters.kusto
#                     ^ source.kusto meta.query.top-hitters.kusto constant.numeric.kusto
#                      ^ source.kusto meta.query.top-hitters.kusto
#                       ^^ source.kusto meta.query.top-hitters.kusto keyword.other.operator.kusto
#                         ^ source.kusto meta.query.top-hitters.kusto
#                          ^^^^^^^ source.kusto meta.query.top-hitters.kusto variable.other.column.kusto
>
>PageViews | top-hitters 25 of Page by NumViews
#^^^^^^^^^^^^ source.kusto
#            ^^^^^^^^^^^ source.kusto meta.query.top-hitters.kusto keyword.other.query.kusto
#                       ^ source.kusto meta.query.top-hitters.kusto
#                        ^^ source.kusto meta.query.top-hitters.kusto constant.numeric.kusto
#                          ^ source.kusto meta.query.top-hitters.kusto
#                           ^^ source.kusto meta.query.top-hitters.kusto keyword.other.operator.kusto
#                             ^ source.kusto meta.query.top-hitters.kusto
#                              ^^^^ source.kusto meta.query.top-hitters.kusto variable.other.column.kusto
#                                  ^ source.kusto meta.query.top-hitters.kusto
#                                   ^^ source.kusto meta.query.top-hitters.kusto keyword.other.operator.kusto
#                                     ^ source.kusto meta.query.top-hitters.kusto
#                                      ^^^^^^^^ source.kusto meta.query.top-hitters.kusto variable.other.column.kusto
>